### ISSUE TITLE
Make Test Fails

### ISSUE THREAD

**Issue by murphymj25** [fabiolb/fabio#591]
When i run a make test, i get the following failure:

```
=== RUN   TestGetCertificate
Issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
2019/01/15 12:56:03 [ERROR] cert: vault: Failed to re-issue cert for localhost: vault: issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
Issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
2019/01/15 12:56:03 [ERROR] cert: vault: Failed to re-issue cert for localhost: vault: issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
Issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
Issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
Issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
2019/01/15 12:56:03 [ERROR] cert: vault: Failed to re-issue cert for localhost: vault: issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
Issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
2019/01/15 12:56:03 [ERROR] cert: vault: Failed to re-issue cert for localhost: vault: issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
2019/01/15 12:56:03 [ERROR] cert: vault: Failed to re-issue cert for localhost: vault: issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
2019/01/15 12:56:03 [ERROR] cert: vault: Failed to re-issue cert for localhost: vault: issue: Put http://127.0.0.1:58421/v1/test-pki/issue/fabio: EOF
--- PASS: TestGetCertificate (0.04s)
FAIL
FAIL	github.com/fabiolb/fabio/cert	8.111s
```
Is there something i need to setup differently for this test to pass?

Thanks!

**Comment by pschultz** [comment]
TestGetCertificate isn't failing:

    --- PASS: TestGetCertificate (0.04s)

Look at the test output again, and find out which one is actually failing. Then run it by itself (with the -run flag), and paste the output here.

**Comment by murphymj25** [comment]
That's the odd thing, i'm not sure what piece is failing since each test says it passes.  The only place i see a fail is those two spots:

FAIL
FAIL	github.com/fabiolb/fabio/cert	8.111s

Thanks!

**Comment by pschultz** [comment]
I can reproduce the failure with Vault 0.9.1 and will look into it. 

Note that 0.9.1 is pretty old. The tests pass with 1.0.2.

**Comment by pschultz** [comment]
With #497 we essentially dropped support for Vault versions before 0.10.1, specifically because we rely on [this change](https://github.com/hashicorp/vault/blob/v0.10.1/CHANGELOG.md) to figure out if the secret backend uses the K/V engine in version 1 or version 2:

> * Mount information visibility: Users that have access to any path within a mount can now see information about that mount, such as its type and options, via some API calls.

Given that 0.10.1 has been released in April 2018, and 1.0 has been released in December 2018, I'm not convinced that it's worth the effort of making this work. I suggest updating to a more recent Vault version instead.

/cc @magiconair, @leprechau

**Comment by murphymj25** [comment]
I updated my local vault to v1.0.2 and the tests now pass.  Thanks everyone for the help!

Closing this issue.
